### ISSUE TITLE
Slider needs to support data-driven and non-numeric values

### ISSUE THREAD

**Issue by marcysutton** [angular/material#200]
To be the most flexible, `material-slider` should support data-driven values and non-numeric values. 

As an example of data-driven: in left-to-right languages, the minimum value is on the left, with the maximum value on the right. To support left-to-right languages, these should be reversed. This shouldn't be too difficult by initializing the slider dynamically from data-driven values and applying the appropriate ARIA states and properties accordingly. 

For non-numeric values, the `aria-valuetext` attribute should be added along with the other slider requirements: `role="slider"`, `aria-valuenow`, `aria-valuemin`, and `aria-valuemax`. 

For more information: http://www.w3.org/TR/wai-aria/states_and_properties#aria-valuenow and https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_slider_role
